hw-model/src/lib.rs

```diff
@@ -528,15 +529,11 @@ pub trait HwModel {
     /// Streams `data` to the sha512acc SoC interface. If `sha384` computes
     /// the sha384 hash of `data`, else computes sha512 hash.
     ///
-    /// `dlen` is the size of data in bytes. sha512acc does not require streamed
-    /// data to be full words.
-    ///
     /// Returns the computed digest if the sha512acc lock can be acquired.
     /// Else, returns an error.
     fn compute_sha512_acc_digest(
         &mut self,
-        data: &[u32],
-        dlen: u32,
+        data: &[u8],
         sha384: bool,
     ) -> Result<Vec<u32>, ModelError> {
```

Comment:
If the input is bytes, maybe the output should be bytes too?

Reply:
Done